packages/block-editor/src/components/block-variation-transforms/index.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { isMatch } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	DropdownMenu,
+	MenuGroup,
+	MenuItemsChoice,
+} from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { chevronDown } from '@wordpress/icons';
+
+const getMatchedVariation = ( blockAttributes, variations ) => {
+	if ( ! variations || ! blockAttributes ) return;
+	return variations.find( ( { attributes } ) => {
+		if ( ! attributes || ! Object.keys( attributes ).length ) return false;
+		return isMatch( blockAttributes, attributes );
+	} );
+};
+
+function __experimentalBlockVariationTransforms( { blockClientId } ) {
+	const [ selectedValue, setSelectedValue ] = useState( '' );
+	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
+	const { variations, blockAttributes } = useSelect(
+		( select ) => {
+			const { getBlockVariations } = select( 'core/blocks' );
+			const { getBlockName, getBlockAttributes } = select(
+				'core/block-editor'
+			);
+			const blockName = blockClientId && getBlockName( blockClientId );
+			return {
+				variations:
+					blockName && getBlockVariations( blockName, 'transform' ),
+				blockAttributes: getBlockAttributes( blockClientId ),
+			};
+		},
+		[ blockClientId ]
+	);
+	useEffect( () => {
+		const matchedVariation = getMatchedVariation(
+			blockAttributes,
+			variations
+		);
+		setSelectedValue( matchedVariation?.name || '' );
```

Comment:
Since we have the helper, we can return the name or `null` when not found:
```suggestion
		setSelectedValue( getMatchingVariationName(
 			blockAttributes,
 			variations
 		) );
```

I updated the name in this example to `getMatchingVariationName`. Is it possible to set `null` as a value to make it explicit that no match was found?

Reply:
>Is it possible to set null as a value to make it explicit that no match was found?

Why `null`. Don't you think `undefined` is okay as well? check: https://github.com/WordPress/gutenberg/pull/26687/commits/853b184d50e2e069db74ad26b4450fc36c69d48c

Reply:
Whatever you prefer.